### PR TITLE
Liquidity rewards

### PR DESCRIPTION
**Pull request by michalsmiarowski**
Depends on: #2224
Depends on: #2231

This PR adds new Liquidity Rewards page.

Main changes:
- add new **Liquidity Rewards** page
- add function that calculates the APY (although it is not used for now, since we are not sure about the results. All the APY results in UI are hardcoded and calculated for $1M pool )
- calculate % of the total pool for each liquidity pair
- calculate KEEP tokens reward for each liquidity pair
- add possibility to add more lp tokens for each liquidity pair
- add possibility to withdraw rewards for each liquidity pair
- santa hats for each card
- **NEW** label in the left menu for **Liquidity** page
- **NEW** label for the Liquidity page header

image

### CODE REVIEW

File: solidity/dashboard/src/components/LiquidityRewardCard.jsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import React from "react"
+import DoubleIcon from "./DoubleIcon"
+import * as Icons from "./Icons"
+import { SubmitButton } from "./Button"
+import Card from "./Card"
+
+const LiquidityRewardCard = ({ title, MainIcon, SecondaryIcon, viewPoolLink }) => {
+  return (
+    <Card className={"tile"}>
+      <div className={"liquidity__card-title-section"}>
```
Comment (r-czajkowski) [inline solidity/dashboard/src/components/LiquidityRewardCard.jsx:10]: At the top in `Card` component should be `liquidity-card` then `liquidity-card__title`(I think the section word is unnecessary here) and so on. According to the BEM methodology.

File: solidity/dashboard/src/components/LiquidityRewardCard.jsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+          SecondaryIcon={SecondaryIcon}
+          className={`liquidity__double-icon-container`}
+        />
+        <h2 className={"h2--alt text-grey-70"}>{title}</h2>
+      </div>
+      <div className={"liquidity-card-subtitle-section"}>
+        <span className="text-grey-40">Uniswap Pool&nbsp;</span>
+        <a
+          href={viewPoolLink}
+          className="arrow-link text-small"
```
Comment (r-czajkowski) [inline solidity/dashboard/src/components/LiquidityRewardCard.jsx:22]: Unnecessary `arrow-link` class. We handle external links via [selector](https://github.com/keep-network/keep-core/blob/master/solidity/dashboard/src/css/typography.less#L105).

File: solidity/dashboard/src/components/LiquidityRewardCard.jsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+          style={{ marginLeft: "auto", marginRight: "2rem" }}
```
Comment (r-czajkowski) [inline solidity/dashboard/src/components/LiquidityRewardCard.jsx:23]: We can drop this `style` prop.

File: solidity/dashboard/src/components/LiquidityRewardCard.jsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      <div className={"liquidity__card-title-section"}>
+        <DoubleIcon
+          MainIcon={MainIcon}
+          SecondaryIcon={SecondaryIcon}
+          className={`liquidity__double-icon-container`}
+        />
+        <h2 className={"h2--alt text-grey-70"}>{title}</h2>
+      </div>
+      <div className={"liquidity-card-subtitle-section"}>
+        <span className="text-grey-40">Uniswap Pool&nbsp;</span>
```
Comment (r-czajkowski) [inline solidity/dashboard/src/components/LiquidityRewardCard.jsx:19]: Based on the Figma views it should be `h4` tag. So I think we can drop `<div className={"liquidity-card-subtitle-section"}>` and:
 
 ```
<h4 className="liquidity-card__subtitle text-grey-40">
 Uniswap Pool&nbsp;
 <a href={viewPoolLink} className="arrow-link text-small">
  View pool
 </a>
</h4>
 ```

File: solidity/dashboard/src/components/LiquidityRewardCard.jsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        </span>
+        <div className={"liquidity__token-balance_values text-grey-70"}>
+          <h3 className={"liquidity__token-balance_values_label"}>
+            <Icons.KeepOutline />
+            <span>KEEP</span>
+          </h3>
+          <h3>1,000,000</h3>
+        </div>
+      </div>
+      <div className={"liquidity__add-more-tokens"}>
```
Comment (r-czajkowski) [inline solidity/dashboard/src/components/LiquidityRewardCard.jsx:50]: I think it's unecessary `div` wrapper. We can pass class name directly to the `SubmitButton` component.

File: solidity/dashboard/src/components/LiquidityRewardCard.jsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        <SubmitButton className={`btn btn-primary btn-lg w-100`}>
+          add more lp tokens
+        </SubmitButton>
+      </div>
+      <div className={"liquidity__withdraw"}>
```
Comment (r-czajkowski) [inline solidity/dashboard/src/components/LiquidityRewardCard.jsx:55]: same as above

File: solidity/dashboard/src/components/LiquidityRewardCard.jsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        <SubmitButton className={"btn btn-primary btn-lg w-100 text-black"}>
```
Comment (r-czajkowski) [inline solidity/dashboard/src/components/LiquidityRewardCard.jsx:56]: should be `btn-secondary` and without `text-black`.

File: solidity/dashboard/src/css/card.less [hunk 8]
```diff
@@ -0,0 +1,6 @@
+.card-root {
+  border: 1px solid black;
```
Comment (r-czajkowski) [inline solidity/dashboard/src/css/card.less:2]: As I can see on Figma the `border` prop is unnecessary here.

File: solidity/dashboard/src/pages/liquidity/LiquidityPage.jsx [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+import PageWrapper from "../../components/PageWrapper"
+import CardContainer from "../../components/CardContainer"
+import LiquidityRewardCard from "../../components/LiquidityRewardCard"
+import { LIQUIDITY_REWARD_PAIR } from "../../constants/constants"
+import * as Icons from "../../components/Icons"
+
+const LiquidityPage = ({ title }) => {
+  return (
+    <PageWrapper title={title}>
+      <CardContainer className={"flex wrap"}>
```
Comment (r-czajkowski) [inline solidity/dashboard/src/pages/liquidity/LiquidityPage.jsx:11]: I think `CardContainer` should be responsible for wrapping elements and we shouldn't worry about it every time we use the` CardContainer` component.

**Review by r-czajkowski — CHANGES_REQUESTED** [review]
Left some notes, overall looks good. Let's run `npm run js:lint:fix` to fix the linting errors. Also, I think we should update the linting rules about passing a string as a prop, because in this PR we wrap the string in `{}` which in my opinion it's unnecessary and makes the code less readable.

File: solidity/dashboard/src/components/Card.jsx [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import React from "react"
+import "../css/card.less"
```
[the remark this answers is not in the document]
Comment (michalsmiarowski) [reply solidity/dashboard/src/components/Card.jsx]: Fixed: https://github.com/keep-network/keep-core/pull/2216/commits/a33c195be2089108393e8b614df2da07295b630b

File: solidity/dashboard/src/components/CardContainer.jsx [hunk 11]
```diff
@@ -0,0 +1,12 @@
+import React from "react"
+import "../css/card-container.less"
```
[the remark this answers is not in the document]
Comment (michalsmiarowski) [reply solidity/dashboard/src/components/CardContainer.jsx]: Fixed: https://github.com/keep-network/keep-core/pull/2216/commits/a33c195be2089108393e8b614df2da07295b630b

File: solidity/dashboard/src/components/LiquidityRewardCard.jsx (hunk 2 above)
Comment (michalsmiarowski) [reply solidity/dashboard/src/components/LiquidityRewardCard.jsx]: Fixed in https://github.com/keep-network/keep-core/pull/2216/commits/0cd07235983324ca719889e5738b09bda0e4dc16

File: solidity/dashboard/src/components/LiquidityRewardCard.jsx (hunk 1 above)
Comment (michalsmiarowski) [reply solidity/dashboard/src/components/LiquidityRewardCard.jsx]: Fixed in https://github.com/keep-network/keep-core/pull/2216/commits/ff520422a7149ced8fa83e2c17182dd05fa9f156

File: solidity/dashboard/src/components/LiquidityRewardCard.jsx (hunk 3 above)
Comment (michalsmiarowski) [reply solidity/dashboard/src/components/LiquidityRewardCard.jsx]: I think this was fixed by you in one of your commits.

File: solidity/dashboard/src/components/LiquidityRewardCard.jsx (hunk 4 above)
Comment (michalsmiarowski) [reply solidity/dashboard/src/components/LiquidityRewardCard.jsx]: Fixed in https://github.com/keep-network/keep-core/pull/2216/commits/bcb0728b0a7808362814406b1cbb02beca04ea7d

File: solidity/dashboard/src/components/LiquidityRewardCard.jsx (hunk 5 above)
Comment (michalsmiarowski) [reply solidity/dashboard/src/components/LiquidityRewardCard.jsx]: Fixed in https://github.com/keep-network/keep-core/pull/2216/commits/c8ec5520a4b2c6745dc0be1ae028e4fb9c1c7eef

File: solidity/dashboard/src/components/LiquidityRewardCard.jsx (hunk 6 above)
Comment (michalsmiarowski) [reply solidity/dashboard/src/components/LiquidityRewardCard.jsx]: Fixed in https://github.com/keep-network/keep-core/pull/2216/commits/c8ec5520a4b2c6745dc0be1ae028e4fb9c1c7eef

File: solidity/dashboard/src/components/LiquidityRewardCard.jsx (hunk 7 above)
Comment (michalsmiarowski) [reply solidity/dashboard/src/components/LiquidityRewardCard.jsx]: Fixed in https://github.com/keep-network/keep-core/pull/2216/commits/c191d060b8a7a7c043cf10f7e6fa014b94691591

File: solidity/dashboard/src/css/card.less (hunk 8 above)
Comment (michalsmiarowski) [reply solidity/dashboard/src/css/card.less]: Fixed in https://github.com/keep-network/keep-core/pull/2216/commits/bd7dbb50d3b769a27468fe3e1e2f252924a8ee04

File: solidity/dashboard/src/pages/liquidity/LiquidityPage.jsx (hunk 9 above)
Comment (michalsmiarowski) [reply solidity/dashboard/src/pages/liquidity/LiquidityPage.jsx]: Fixed in https://github.com/keep-network/keep-core/pull/2216/commits/b0b960c8b0fbc53620cacdef5730474e8980c2a6

**Comment by michalsmiarowski** [comment]
@r-czajkowski I think i fixed all the issues so you can check it again

File: solidity/dashboard/src/css/double-icon.less [hunk 12]
```diff
@@ -0,0 +1,13 @@
+.double-icon-container {
+  display: flex;
+  flex-direction: row-reverse;
+
+  .main-icon, .secondary-icon {
```
Comment (r-czajkowski) [inline solidity/dashboard/src/css/double-icon.less:5]: Based on the Figam views, `main-icon` should have `filter: drop-shadow(2px 0px 4px rgba(0, 0, 0, 0.25));` prop.

**Review by r-czajkowski — CHANGES_REQUESTED** [review]
Left some thoughts. Want to flag that we should add background to the tbtc svg icon since is not looking good if the `tbtc` icon is a `MainIcon` in `DoubleIcon` component.

obraz

**Review by r-czajkowski — APPROVED** [review]
Let's ship it :shipit:

**Pull request merged**